src/main/sensors/gyro.c

```diff
@@ -422,8 +443,78 @@ bool gyroInit(void)
         gyroDebugMode = DEBUG_NONE;
         break;
     }
+
+    bool ret = false;
     memset(&gyro, 0, sizeof(gyro));
-    return gyroInitSensor(&gyroSensor1);
+    gyroToUse = gyroConfig()->gyro_to_use;
+
+#if defined(USE_DUAL_GYRO) && defined(GYRO_1_CS_PIN)
+    if (gyroToUse == GYRO_CONFIG_USE_GYRO_1 || gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH) {
+        gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin = IOGetByTag(IO_TAG(GYRO_1_CS_PIN));
+        IOInit(gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin, OWNER_MPU_CS, 0);
+        IOHi(gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin); // Ensure device is disabled, important when two devices are on the same bus.
+        IOConfigGPIO(gyroSensor1.gyroDev.bus.busdev_u.spi.csnPin, SPI_IO_CS_CFG);
+    }
+#endif
+
+#if defined(USE_DUAL_GYRO) && defined(GYRO_2_CS_PIN)
+    if (gyroToUse == GYRO_CONFIG_USE_GYRO_2 || gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH) {
+        gyroSensor2.gyroDev.bus.busdev_u.spi.csnPin = IOGetByTag(IO_TAG(GYRO_2_CS_PIN));
+        IOInit(gyroSensor2.gyroDev.bus.busdev_u.spi.csnPin, OWNER_MPU_CS, 0);
```

Comment:
Last argument should be `1` to reflect that fact that it's different from `GYRO_1_CS_PIN`

Reply:
👍